Kudu.Services/Diagnostics/CrashDumpController.cs

```diff
@@ -0,0 +1,121 @@
+        {
+            using (_tracer.Step($"CrashDumpController.Delete({name})"))
+            {
+                var crashDump = GetCrashDump(name);
+                FileSystemHelpers.DeleteFile(crashDump.FilePath);
+                return Request.CreateResponse(HttpStatusCode.OK);
+            }
+        }
+
+        [HttpPost]
+        public async Task<HttpResponseMessage> Analyze(string name)
+        {
+            using (_tracer.Step($"CrashDumpController.Analyze({name})"))
+            {
+                var crashDump = GetCrashDump(name);
+                var exe = new Executable(ResolveCdbPath(), _environment.RootPath, _settings.GetCommandIdleTimeout());
+                using (MemoryStream outputStream = new MemoryStream(), errorStream = new MemoryStream())
+                {
+                    // ExecuteAsync could deadlock if called from an ASP.NET thread
+                    var exitCode = await Task.Run(async () => await exe.ExecuteAsync(_tracer, $"-c \"!analyze -v;q\" -z \"{crashDump.FilePath}\"", outputStream, errorStream));
+                    if (exitCode == 0)
+                    {
+                        return Request.CreateResponse(HttpStatusCode.OK, new { output = outputStream.AsString() });
+                    }
+                    else
+                    {
+                        return Request.CreateResponse(HttpStatusCode.InternalServerError, new { error = errorStream.AsString() });
+                    }
+                }
+            }
+        }
+
+        //public for unit tests
+        public CrashDumpInfo GetCrashDump(string name)
+        {
+            var path = Path.Combine(_environment.CrashDumpsPath, name);
+            if (FileSystemHelpers.FileExists(path))
+            {
+                var fileInfo = FileSystemHelpers.FileInfoFromFileName(path);
+                return GetCrashDumpInfoFromFile(fileInfo, Request.RequestUri.GetLeftPart(UriPartial.Authority));
+            }
+            throw new HttpResponseException(HttpStatusCode.NotFound);
+        }
+
+        private static CrashDumpInfo GetCrashDumpInfoFromFile(FileInfoBase fileInfo, string baseUrl)
+        {
+            baseUrl = baseUrl.TrimEnd('/');
+            return new CrashDumpInfo
+            {
+                Name = fileInfo.Name,
+                Timestamp = fileInfo.LastWriteTime,
+                Href = new Uri($"{baseUrl}/api/crashdumps/{fileInfo.Name}"),
+                AnalyizeHref = new Uri($"{baseUrl}/api/crashdumps/{fileInfo.Name}/analyze"),
+                DownloadHref = new Uri($"{baseUrl}/api/vfs/data/{Constants.Dumps}/{fileInfo.Name}"),
+                FilePath = fileInfo.FullName
+            };
+        }
+
+        private static string ResolveCdbPath()
+        {
+            var programFiles = System.Environment.GetFolderPath(System.Environment.SpecialFolder.ProgramFilesX86);
+            return Path.Combine(programFiles, "Windows Kits", "8.1", "Debuggers", "x64", "cdb.exe");
```

Comment:
I don't think we get to pick the name of the dumps. At least I couldn't find any procdump switch for that would allow that. But I can make them go into different folders. i.e: `\dumps\x64` and `\dumps\x86`